Mapbox/MapboxMapsStyle/Style.swift

```diff
@@ -367,6 +388,48 @@ public class Style {
             return .failure(.addLightFailed(nil))
         }
     }
+
+    /// Updates a layer that exists in the style already
+    /// - Parameters:
+    ///   - id: identifier of layer to update
+    ///   - type: Type of the layer
+    ///   - update: Closure that mutates a layer passed to it
+    /// - Returns: Result type with  `.success` if update is successful, `LayerError` otherwise
+    @discardableResult
+    func updateLayer<T: Layer>(id: String, type: T.Type, update: (inout T) -> Void) -> Result<Bool, LayerError> {
+
+        let result = self.getLayer(with: id, type: T.self)
+        var layer: T?
+
+        // Fetch the layer from the style
+        switch result {
+        case .success(let retrievedLayer):
+            // Successfully retrieved the layer
+            layer = retrievedLayer
+        case .failure(_):
+
+            // Could not retrieve the layer
+            return .failure(.getStyleLayerFailed(nil))
+        }
+
+        guard var validLayer = layer else {
+            return .failure(.retrievedLayerIsNil)
+        }
+
+        // Call closure to update the retrieved layer
+        update(&validLayer)
+
```

Comment:
Do we need to save the updated layer to the cache?

Reply:
whoops, thought I was doiign that.

Reply:
Got rid of layer cache for now.